WordPress/Classes/Networking/BlogServiceRemoteREST.m

```diff
@@ -118,6 +119,89 @@ - (void)syncConnectionsForBlog:(Blog *)blog
           }];
 }
 
+- (void)checkAuthorizationForPublicizer:(Publicizer *)service
+                                success:(AuthorizationHandler)success
+                                failure:(void (^)(NSError *))failure
+{
+    NSParameterAssert([service isKindOfClass:[Publicizer class]]);
+    NSParameterAssert(service.blog.dotComID != nil);
+    
+    NSString *path = @"me/keyring-connections";
+    NSString *requestUrl = [self pathForEndpoint:path
+                                     withVersion:ServiceRemoteRESTApiVersion_1_1];
+    
+    [self.api GET:requestUrl
+       parameters:nil
+          success:^(AFHTTPRequestOperation *operation, id responseObject) {
+              NSArray *keyrings = [responseObject arrayForKey:@"connections"];
+              for (NSDictionary *keyring in keyrings) {
+                  if ([[keyring stringForKey:@"service"] isEqualToString:service.service]) {
+                      if (success) {
+                          success(keyring);
+                      }
+                      return;
+                  }
+              }
+              if (failure) {
+                  failure(nil);
+              }
+          } failure:^(AFHTTPRequestOperation *operation, NSError *error) {
+              if (failure) {
+                  failure(error);
+              }
+          }];
+}
+
+- (void)connectPublicizer:(Publicizer *)service
+        withAuthorization:(NSDictionary *)authorization
+                  success:(SuccessHandler)success
+                  failure:(void (^)(NSError *))failure
+{
+    NSParameterAssert([service isKindOfClass:[Publicizer class]]);
+    NSParameterAssert(service.blog.dotComID != nil);
+    
+    NSString *path = [self pathForConnectionWithPublicizer:service];
+    NSString *requestUrl = [self pathForEndpoint:path
+                                     withVersion:ServiceRemoteRESTApiVersion_1_1];
+    NSDictionary *parameters = @{ @"keyring_connection_ID" : authorization[@"ID"] };
```

Comment:
Comment/Suggestion: since the only two callers that use this method just pass the `ID` parameter, can we replace the `Authorization` dictionary with an explicit parameter?.